node/src/blockchain/blockchain_interface/blockchain_interface_web3/mod.rs

```diff
@@ -1051,40 +642,33 @@ mod tests {
 
     #[test]
     fn blockchain_interface_web3_can_build_blockchain_agent() {
-        let get_transaction_fee_balance_params_arc = Arc::new(Mutex::new(vec![]));
-        let get_masq_balance_params_arc = Arc::new(Mutex::new(vec![]));
-        let get_transactions_id_params_arc = Arc::new(Mutex::new(vec![]));
+        let port = find_free_port();
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            // gas_price
+            .response("0x3B9ACA00".to_string(), 0) // 1000000000
```

Comment:
This is due to the fact that we are rounding off. Its protects the case were values are smaller that 1000000000.
So currently we are added +1 Gwei when getting the gas price.

We did this as a temporary measure. should be fixed when we add full support for wei

Reply:
But do you understand that these conditions make the tests fragile and misleading. You may remember these little caveats (actually just pure hacks), but once this is merged with master, it will immediately become a potential trap for others. I don't really the idea of hot-fixing things like this. 


To me it speaks like there are grounded arguments for you guys to change the units in this part of the code, even in a wider area, still I don't think I'm talking of more than the paying mechanism, its cascade, that consists of the bill blocks like:

BlockchainAgent ---> MAKE IT STORE THE ON-CHAIN DETERMINED GAS PRICE IN WEI, NOT GWEI! (Including changes in its methods).

PaymentAdjuster --- In your card it is still just an empty vessel. You don't have anything to worry about there.

The final stage with the batch paying ---> YOU MUST CHANGE IT HERE TOO, BUT THE COMPILER WILL BE YOUR FRIEND ON THE JOURNEY. IT'S GOING TO BE FAIRLY EASY.

FIX TESTS AFTERWARDS.

EXPECT MAYBE SOME LITTLE ISSUES FROM SOME PARTS OF THAT CODE SHARED WITH THE MULTINODE TESTS. FIX THIS CRATE LATER OR IMMEDIATELY, AGAIN, THE COMPILER WILL BE YOUR FRIEND.

**URGENCY STATUS: I DO BELIEVE THIS ALL IS DOABLE AND VITAL. IT NEEDS TO BE CARRIED OUT.**  
This card has already shown some ambition of setting the gas price straight by taking in an oracle, but it would be naive to think it can be implemented only partially and we'd ignore the issues that emerge elsewhere because of it. This action I'm talking of isn't actually such a big change. You also shouldn't be blocked by any puzzles. This is going to be just routine coding.

Reply:
Completed :)